ansible/roles_ocp_workloads/ocp4_workload_codeready_workspaces/tasks/remove_workload.yml

```diff
@@ -0,0 +1,78 @@
+---
+# Implement your Workload removal tasks here
+
+- name: Remove CodeReady Workspaces Installation
+  k8s:
+    state: absent
+    api_version: org.eclipse.che/v1
+    kind: CheCluster
+    name: codeready-workspaces
+    namespace: "{{ ocp4_workload_codeready_workspaces_namespace }}"
+
+- name: Wait until all CodeReady Workspaces pods have been removed
+  k8s_info:
+    api_version: v1
+    kind: Pod
+    namespace: "{{ ocp4_workload_codeready_workspaces_namespace }}"
+  register: r_crw_pods
+  retries: 20
+  delay: 5
+  until: r_crw_pods.resources | length <= 2
+  # The Operator pod will remain the only running pod
+  # For catalog mirrors there will be the catalog pod as well
+
+- name: Get Installed CSV
+  k8s_info:
+    api_version: operators.coreos.com/v1alpha1
+    kind: Subscription
+    name: codeready-workspaces
+    namespace: "{{ ocp4_workload_codeready_workspaces_namespace }}"
+  register: r_subscription
+
+- name: Remove CSV
+  when:
+  - r_subscription.resources | length > 0
+  - r_subscription.resources[0].status.currentCSV is defined
+  - r_subscription.resources[0].status.currentCSV | length > 0
+  k8s:
+    state: absent
+    api_version: operators.coreos.com/v1alpha1
+    kind: ClusterServiceVersion
+    name: "{{ r_subscription.resources[0].status.currentCSV }}"
+    namespace: "{{ ocp4_workload_codeready_workspaces_namespace }}"
+
+- name: Remove Subscription
+  k8s:
+    state: absent
+    definition: "{{ lookup('template', item ) | from_yaml }}"
+  loop:
+  - ./templates/subscription.j2
```

Comment:
Why the loop for this? Won't this just remove a single Subscription called `codeready-workspaces` since that is what the template that is used to deploy is hardcodced as?

It is easier to read at a glance for a simple removal to just have it inline IMO.

Reply:
Fine.